x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/Security.java

```diff
@@ -303,19 +307,15 @@ public Security(Settings settings, final Path configPath) {
         this.env = new Environment(settings, configPath);
         this.enabled = XPackSettings.SECURITY_ENABLED.get(settings);
         if (enabled) {
-            runStartupChecks(settings);
+            runStartupChecks(settings, getLicenseState());
```

Comment:
Is license state meaningful at this point during node start up?

Reply:
I see your point. Should we just remove checks for license on node startup and only leave `ValidateLicenseForFIPS` on node joins ?